src/test/index.test.ts

```diff
@@ -1,3 +1,256 @@
+      client.rest.pulls.listFiles.mockResolvedValue({ data: [{ filename: "README.md" }] });
+      pr.user.login = "gavofyork";
+      const result = await runner.runAction({ configLocation: "abc" });
+      expect(result.reports).toHaveLength(1);
+      expect(result.conclusion).toBe("failure");
+      const report = getReport(result.reports, "Core developers");
+      console.log(report);
+      expect(report.missingReviews).toBe(1);
+    });
+
+    test("should approve PR if it has enough approvals", async () => {
+      // @ts-ignore
+      client.rest.pulls.listFiles.mockResolvedValue({ data: [{ filename: "README.md" }] });
+      mockReviews([
+        { login: "core-1", state: "approved", id: 12 },
+        { login: "core-2", state: "approved", id: 123 },
+      ]);
+      const result = await runner.runAction({ configLocation: "abc" });
+      expect(result.reports).toHaveLength(0);
+      expect(result.conclusion).toBe("success");
+    });
+  });
+
+  test("should request a runtime upgrade review if the file is from runtime upgrades", async () => {
+    // @ts-ignore
+    client.rest.pulls.listFiles.mockResolvedValue({ data: [{ filename: "cumulus/parachains/common/src/example.rs" }] });
+
+    const result = await runner.runAction({ configLocation: "abc" });
+    expect(result.reports).toHaveLength(1);
+    expect(result.conclusion).toBe("failure");
+    const report = getReport(result.reports, "Runtime files cumulus");
+    expect(report.missingReviews).toBe(2);
+  });
+
+  test("should request only one runtime upgrade review if the file is from runtime upgrades and the author belongs to one of the teams", async () => {
+    // @ts-ignore
+    client.rest.pulls.listFiles.mockResolvedValue({ data: [{ filename: "cumulus/parachains/common/src/example.rs" }] });
+    pr.user.login = "gavofyork";
+    const result = await runner.runAction({ configLocation: "abc" });
+    expect(result.reports).toHaveLength(1);
+    expect(result.conclusion).toBe("failure");
+    const report = getReport(result.reports, "Runtime files cumulus");
+    console.log(report);
+    expect(report.missingReviews).toBe(2);
+  });
+
+  describe("Combinations", () => {
+    test("should use same reviewer for separate rules", async () => {
+      client.rest.pulls.listFiles.mockResolvedValue({
+        // @ts-ignore
+        data: [{ filename: "cumulus/parachains/common/src/example.rs" }, { filename: "README.md" }],
+      });
+      mockReviews([{ state: "approved", id: 123, login: "gavofyork" }]);
+      const newResult = await runner.runAction({ configLocation: "abc" });
+      console.log("nre result", JSON.stringify(newResult, null, 2));
+      expect(newResult.reports.map((r) => r.missingReviews).reduce((a, b) => a + b, 0)).toBe(3);
+    });
+
+    test("should use same reviewers for separate rules", async () => {
+      client.rest.pulls.listFiles.mockResolvedValue({
+        // @ts-ignore
+        data: [{ filename: "" }, { filename: "README.md" }],
```

Comment:
do you think would be good to have here some runtime file changed, so it involves locked-review?

Reply:
It was the intention, I seemed to have forgotten to paste it